extension/landing-zone/variables.tf

```diff
@@ -0,0 +1,328 @@
+variable "ibmcloud_api_key" {
+  description = "The API key that's associated with the account to provision resources to"
+  type        = string
+  sensitive   = true
+}
+
+variable "resource_group" {
+  type        = string
+  description = "An existing resource group name to use for this example."
+}
+
+variable "region" {
+  description = "The region of the existing LZ VPC."
+  type        = string
+}
+
+variable "prefix" {
+  description = "The prefix that you would like to append to VSI, Block Storage, Security Group, Floating IP and Load Balancer."
+  type        = string
+  default     = "slz-vsi"
+}
+
+variable "vpc_id" {
+  description = "The ID of the VPC where the VSI is created."
+  type        = string
+}
+
+variable "ssh_keys" {
+  description = "SSH keys to use to provision a VSI. Must be an RSA key with a key size of either 2048 bits or 4096 bits (recommended). If `public_key` is not provided, the named key will be looked up from data. See https://cloud.ibm.com/docs/vpc?topic=vpc-ssh-keys."
+  type = object({
+    name       = string
+    public_key = optional(string)
+  })
+
+  validation {
+    error_message = "Public SSH Key must be a valid ssh rsa public key."
+    condition     = var.ssh_keys.public_key == null || can(regex("ssh-rsa AAAA[0-9A-Za-z+/]+[=]{0,3} ?([^@]+@[^@]+)?", var.ssh_keys.public_key))
+  }
+}
+
+variable "resource_tags" {
+  description = "List of tags for the VSI, Block Storage, Security Group, Floating IP and Load Balancer created"
+  type        = list(string)
+  default     = null
+}
+
+variable "access_tags" {
+  type        = list(string)
+  description = "A list of access tags to apply to the VSI resources created by the module."
+  default     = []
+}
+
+variable "image_name" {
+  description = "Image ID used for VSI. Run 'ibmcloud is images' to find available images. Be aware that region is important for the image since the id's are different in each region."
```

Comment:
What does this mean?

> Be aware that region is important for the image since the id's are different in each region.

Image IDs are the same in all regions?

Reply:
Yea, the image IDs are different for different regions.
`r006-5f532555-0d74-4749-8bd7-7f7406c87820  <- ibm-ubuntu-22-04-2-minimal-amd64-1 `  us-south
`r038-a55ccf04-916e-408f-a565-58a09947f54e   ibm-ubuntu-22-04-2-minimal-amd64-1 ` ca-tor

Reply:
Ah sorry yes, image IDs are different. But the var name you have here is name, not ID. Should it be ID or name?

Reply:
Here we have the name because there is a data block which fetches the image id using the image_name